archive/app/controllers/ArchiveController.scala

```diff
@@ -40,17 +39,17 @@ class ArchiveController(redirects: RedirectService) extends Controller with Logg
       path match {
         case Gallery(gallery)      => redirectTo(gallery, "gallery")
         case Century(century)      => redirectTo(century, "century")
-        case Guardian(endOfUrl)    => redirectTo(s"www.theguardian.com/$endOfUrl", "guardian")
+        case Guardian(endOfUrl)    => redirectTo(endOfUrl, "guardian")
         case Lowercase(lower)      => redirectTo(lower, "lowercase")
 
         // Googlebot hits a bunch of really old combiners and combiner RSS
         // bounce these to the section
         case CombinerSectionRss(section)                => redirectTo(s"$section/rss", "combinerrss")
         case CombinerSection(section)                   => redirectTo(section, "combinersection")
         case Combiner(combiner)                         => redirectTo(combiner, "combiner")
-        case DatedSpecialIndexPage(section, rest, _)    => cachedArchiveRedirect(WithoutRevalidationResult(Redirect(s"${LinkTo(section)}$rest/all", redirectHttpStatus)))
-        case SectionSpecialIndex(section, _)            => cachedArchiveRedirect(WithoutRevalidationResult(Redirect(s"${LinkTo(section)}/all", redirectHttpStatus)))
-        case NewspaperPage(paper, date, book)           => cachedArchiveRedirect(WithoutRevalidationResult(Redirect(s"${LinkTo(paper)}/$book/$date/all", redirectHttpStatus)))
+        case DatedSpecialIndexPage(section, rest, _)    => redirectTo(section, "datedspecialindexpage", rest, "all")
+        case SectionSpecialIndex(section, _)            => redirectTo(section, "sectionspecialindex", "all")
+        case NewspaperPage(paper, date, book)           => redirectTo(paper, "newspaperpage", book, date, "all")
```

Comment:
because of editionalisation oddities I wasn't able to do `redirectTo(s"$paper/$book/$date/all", "newspaperpage")` (for example)